core/client/fs/src/main/java/alluxio/client/file/cache/PageStore.java

```diff
@@ -43,10 +39,29 @@ public interface PageStore extends AutoCloseable {
   Logger LOG = LoggerFactory.getLogger(PageStore.class);
 
   /**
-   * Creates a new {@link PageStore}. Previous state in the samme cache dir will be overwritten.
+   * Creates a new {@link PageStore}. Previous state in the same cache dir will be overwritten.
    *
    * @param options the options to instantiate the page store
    * @return a PageStore instance
+   */
+  static PageStore openOrCreatePageStore(PageStoreOptions options) {
+    PageStore pageStore;
+    try {
+      pageStore = PageStore.open(options);
+    } catch (IOException e) {
+      try {
+        pageStore = PageStore.create(options);
```

Comment:
create calls `initialize` then `open` and it seems that `initialize` just deletes the directory. So it makes more sense to just check whether the directory exist, if yes, call create, otherwise call open. It's not clear to me if `open` throws exception, whether `initialize` would make those go away. The code is very hard to read.